Sources/StreamChat/Query/Filter+Predicate.swift

```diff
@@ -0,0 +1,92 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+extension Filter {
+    var predicate: NSPredicate? {
+        guard let op = FilterOperator(rawValue: self.operator) else { return nil }
+
+        let unableToResolve: () -> NSPredicate? = {
+            let message = """
+            The filter passed to the query could not be resolved at runtime.
+            Please check the documentation at: https://getstream.io/chat/docs/sdk/ios/client/controllers/channels/#%EF%B8%8F-when-the-filter-cannot-be-auto-resolved
+            Otherwise you can have issues when multiples channel list controllers are alive at the same time
+            """
+            log.assertionFailure(message)
+            return nil
+        }
+
+        let dtoKey = autoKey?.dtoKey ?? key
+
+        let makeComparablePredicate: (_ compareToken: String) -> NSPredicate? = { compareToken in
+            guard let key = dtoKey else { return unableToResolve() }
+            return NSPredicate(format: "\(key) \(compareToken) %@", argumentArray: [value])
+        }
+
+        let makeCompoundPredicate: (_ logicalType: NSCompoundPredicate.LogicalType) -> NSPredicate? = { logicalType in
+            guard let value = self.value as? [Filter] else { return unableToResolve() }
```

Comment:
`self` is strongly captured here, which become dangerous. Maybe consider capturing self weakly. That should change much for the code as it already guards against nil or for an invalid type